### PR TITLE
Second Parser PR

### PR DESCRIPTION
**Pull request by Hetorx**
Please, check my code.

### CODE REVIEW

File: app/second_parser.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+def parse_duration(sec: int) -> str:
+    if sec == 0:
+        raise ValueError ("Given value cannot be equal 0!")
+    if sec < 0:
+        raise ValueError ("Given value cannot be less than 0!")
```
Comment (mateusz2173) [inline app/second_parser.py:5]: Hmmm maybe in case `sec == 0` we should return `0s`, what do you think?

File: app/second_parser.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    days = sec // 86400
+    hours = (sec - days * 86400) // 3600
+    minutes = (sec - days * 86400 - hours * 3600) // 60
+    seconds = sec - days * 86400 - hours * 3600 - minutes * 60
+    result = ("{}d ".format(days) if days else "") + \
+    ("{}h ".format(hours) if hours else "") + \
+    ("{}m ".format(minutes) if minutes else "") + \
+    ("{}s ".format(seconds) [:-1] if seconds else "")
```
Comment (mateusz2173) [inline app/second_parser.py:13]: Nice :)

File: app/tests_parser.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import pytest
+import second_parser
+
+def test_of_convertion():
+    assert second_parser.parse_duration (50) == '50s'
+    assert second_parser.parse_duration (250) == '4m 10s'
+    assert second_parser.parse_duration (3601) == '1h 1s'
+    assert second_parser.parse_duration (86470) == '1d 1m 10s'
```
Comment (mateusz2173) [inline app/tests_parser.py:8]: add here also these tests. Your code doesn't work with these 
```python
    assert parse_duration(1) == "1s"
    assert parse_duration(60) == "1m"
    assert parse_duration(3600) == "1h"
    assert parse_duration(86400) == "1d"

```

File: app/second_parser.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+def parse_duration(sec: int) -> str:
```
Comment (mateusz2173) [inline app/second_parser.py:1]: Move this file to `/app/utility/`

File: app/tests_parser.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import pytest
```
Comment (mateusz2173) [inline app/tests_parser.py:1]: Move this file to `app/tests`

**Review by mateusz2173 — CHANGES_REQUESTED** [review]
Good job 👍 

Minor fixes left to do.
Also don't forget to format your code with `black` and check your files with `pylint` - it could give you hints how to improve some parts of the code (for example string formatting) and is needed for our workflow

also change PR title to something more descriptive

**Review by xNykram — APPROVED** [review]
Approve. However first please adjust changes suggested by Mateusz.

File: app/utility/second_parser.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        and not minutes
+        and days
+        and not seconds
+        or hours
+        and not minutes
+        or hours
+        and not seconds
+        or minutes
+        and not seconds
+    ):
```
Comment (mateusz2173) [inline app/utility/second_parser.py:33]: Hmm this `if` statement is hard to read and I think there is a nicer solution for this. You can remove it and  always return `result[:-1]` and make sure other cases will work with this.

**Review by mateusz2173 — CHANGES_REQUESTED** [review]
- We don't need python cache in remote repo(`.pyc` files), remove them from the remote branch.
- change PR title to something more descriptive and resolve my latest comment: :)
- Add empty `__init__.py` to utility folder
- Resolve my latest comment

Everything else is nice, tests are perfect :)

File: app/utility/second_parser.py (hunk 6 above)
Comment (Hetorx) [reply app/utility/second_parser.py:33]: Yes, it doesn't look so nice, but simple `result[:-1]` without `if` statement is not working as it should. I don't have any other idea how to make it works without modifying whole code.

Comment (mateusz2173) [reply app/utility/second_parser.py:33]: Ok so `result[:-1]` will remove last character from resulting string. Just ensure it will be a space.

**Review by Hetorx — COMMENTED** [review]
done

File: app/utility/second_parser.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    minutes = (sec - days * 86400 - hours * 3600) // 60
+    seconds = sec - days * 86400 - hours * 3600 - minutes * 60
+    result = (
+        (f"{int(days)}d " if days else "")
+        + (f"{int(hours)}h " if hours else "")
+        + (f"{int(minutes)}m " if minutes else "")
+        + (f"{int(seconds)}s " if seconds else "")
+    )
+
+    return result.rstrip()
```
Comment (mateusz2173) [inline app/utility/second_parser.py:17]: nice!

**Review by mateusz2173 — APPROVED** [review]
Good job!

Approving, but please remove .DS_Store files before merging

**Pull request merged**